lib/request-utils.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const { stringify } = require('querystring');
+const RequestError = require('./request-error');
+
+module.exports = class RequestUtils {
+
+	/**
+	 * Converts CallOptions in a valid http/https request options
+	 *
+	 * @static
+	 * @param {CallOptions} {
+	 * 		endpoint, headers, pathParams, queryParams = {}, method = 'GET', path = ''
+	 * 	}
+	 * @returns {object}
+	 */
+	static getOptions({
+		endpoint, headers, pathParams, queryParams = {}, method = 'GET', path = ''
+	}) {
+
+		const formattedEndpoint = this.formatEndpoint(endpoint);
+		const url = new URL(formattedEndpoint);
+
+		const {
+			protocol, host, port
+		} = url;
+
+		const optionsPath = this.getOptionsPath(path, pathParams, queryParams, url);
+
+		return {
+			host,
+			method,
+			path: optionsPath,
+			port,
+			protocol,
+			headers
+		};
+	}
+
+	/**
+	 * Removes protocol from the specified endpoint
+	 *
+	 * @static
+	 * @param {string} endpoint
+	 * @returns {string}
+	 */
+	static formatEndpoint(endpoint) {
+
+		const hasEndpointProtocol = /^https?:\/\//.test(endpoint);
+
+		return hasEndpointProtocol ? endpoint : `http://${endpoint}`;
+	}
+
+	/**
+	 * Resolves http/https request option path
+	 *
+	 * @static
+	 * @param {string} path CallOptions path
+	 * @param {object} pathParams CallOptions pathParams
+	 * @param {object} queryParams CallOptions queryParams
+	 * @param {URL} url
+	 * @returns {string}
+	 */
+	static getOptionsPath(path, pathParams, queryParams, url) {
+
+		const slashRegex = new RegExp('^/');
+
+		const { pathname, searchParams } = url;
+
+		const formattedPathName = decodeURIComponent(pathname.replace(slashRegex, '',));
+		const urlPath = path.replace(slashRegex, '') || formattedPathName;
```

Comment:
```suggestion
		const urlPath = path.replace(slashRegex, '') || decodeURIComponent(pathname.replace(slashRegex, '',));
```
o
```suggestion
		const urlPath = (path || decodeURIComponent(pathname)).replace(slashRegex, '',));
```

Sino, quizás estamos armando `formattedPathName` y no se usa..